src/movies.js

```diff
@@ -1,15 +1,88 @@
 // Iteration 1: All directors? - Get the array of all directors.
+function getAllDirectors(movies) {
+    const allDirectors = movies.map(elm => {
+        return elm.director
+    })
+    return allDirectors
+}
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors. How could you "clean" a bit this array and make it unified (without duplicates)?
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
+function howManyMovies(movies) {
+    const allStevenFilms = movies.filter(elm => {
+        return elm.director.includes('Steven Spielberg') && elm.genre.includes ('Drama')
+    })
+    return allStevenFilms.length
+}
 
 // Iteration 3: All rates average - Get the average of all rates with 2 decimals
+function ratesAverage(movies) {
+    const sumRates = movies.reduce((acc, elm) => acc + elm.rate, 0)
+    const avrRates = sumRates / movies.length
+    const avrFloat = parseFloat(avrRates.toFixed(2))
+    console.log(avrFloat)
+    return avrFloat
```

Comment:
```suggestion
  if (!movies.length) {
        return 0
    }
    const sumRates = movies.reduce((acc, elm) => {
        if (elm.rate) {
            return acc + elm.rate
        } else { 
            return acc 
        }
    }, 0)
    const avrRates = sumRates / movies.length
    const avrFloat = parseFloat(avrRates.toFixed(2))
    console.log(avrFloat)
    return avrFloat
```
Como lo mencionas  faltaban condicionales en caso de no tener `movies` y en el caso de no tener `rate`.